core/src/main/java/org/elasticsearch/index/reindex/AbstractBulkByScrollRequest.java

```diff
@@ -152,7 +152,7 @@ public ActionRequestValidationException validate() {
                             + size + "]",
                     e);
         }
-        if (searchRequest.source().slice() != null && slices != 1) {
+        if (searchRequest.source().slice() != null && !slices.equals(Slices.DEFAULT)) {
```

Comment:
We prefer to use `== false` for negation because it is easier to see visually. You will find this pattern in both Elasticsearch and Lucene.